### ISSUE TITLE
Update / write the description of our images on dockerhub

### ISSUE THREAD

**Issue by nthiery** [sagemath/docker-images#5]
In particular, explore how to automatically update those image descriptions on dockerhub from files maintained in this github repository.

**Comment by embray** [comment]
So previously we discussed that it would be ideal if image descriptions were stored in the image itself (they would be defined in the Dockerfile).

There is no Dockerfile command for adding a description.  Instead the `LABEL` command was added which allows one to attach arbitrary metadata to an image as key/value pairs.  Probably the only reason the `MAINTAINER` command exists is that it predates the `LABEL` command by some time--I suppose it's probably also one of the more important bits of metadata.

Anyways in principle one can store short/long descriptions as labels.  What does not currently exist is a way for Docker Hub to read this info.  It wouldn't be a bad idea to suggest, I think.

In the meantime, what it _does_ support is reading a description out of a README file in a repository when automatic image builds are triggered by push to a git repository.  So for now setting up automated builds would be the way to go.

If we want to put description metadata in the image, _and_ eliminate duplication, we could probably easily include a small script that generates the README for an image from the metadata.

**Comment by embray** [comment]
Note, I've also added docker/hub-feedback#576 (already 11 days ago?? wow, time flies...)

So far it hasn't gotten any attention.  I think it's a fairly obvious thing to do, and I may agitate more for it later.  Not a high priority though.

**Comment by theaverageguy** [comment]
I would like to work on this. A little help please.

**Comment by saraedum** [comment]
I am deploying docs with the approach in #13. I am not building the images on docker hub (because their resources are too limited and their build system is quite minimal). So, I disabled the actual build there (through a hook in the repository) and only deploy the README. You can see the result here: https://hub.docker.com/r/saraedum/sagemath/

**Comment by abitrolly** [comment]
@saraedum so do you mean that full description at https://hub.docker.com/r/saraedum/sagemath/ is updated automatically?

**Comment by saraedum** [comment]
Yes, everything except the one line summary on the top. That can only be changed through the web interface I think.

**Comment by abitrolly** [comment]
@saraedum is there a script that updates description that is runs on CircleCI machine. I don't see any of it.

**Comment by saraedum** [comment]
I don't understand. CirceCI does not update the README on Docker Hub. Docker Hub's automatic build updates the README whenever it detects a change on github.

**Comment by abitrolly** [comment]
@saraedum yes, I thought that there is a way to push README update, and in your case it is still DockerHub that pulls the README itself.

**Comment by YAMLcase** [comment]
@saraedum Would you share your webhook for disabling the build and only allowing the README to be updated?  I'm working with Windows Containers which are not yet supported in Docker Hub so I need a way to just rebuild the doku.  thanks!

**Comment by YAMLcase** [comment]
One option I found that doesn't use webhooks, just regular hooks: put a blank `build` and `push` file in your `./hooks/` directory.  Example I have here that works well: https://github.com/yamlCase/docker-steamcmd-wincore

I'm curious to also know saraedum's solution.

**Comment by saraedum** [comment]
@yamlCase the sagemath repository has this empty hooks approach. It is not enabled yet because I have not told dockerhub to "build" automatically, yet.
